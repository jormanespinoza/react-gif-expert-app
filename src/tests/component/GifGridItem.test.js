import React from "react";
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Testing GifGridItem', () => {
    const gif = {
        title: 'test',
        url: 'test_url'
    }
    const wrapper = shallow(<GifGridItem {...gif} />);

    test('should render <GifGridItem /> properly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should placed gif title parameter into p tag text', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(gif.title);
    });

    test('should placed gif url and title into img tags src and src attributes', () => {
        const img = wrapper.find('img');
        expect(img.props().src).toBe(gif.url);
        expect(img.props().alt).toBe(gif.title);
    });

    test('should have class animated__fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes('animate__fadeIn')).toBeTruthy();
    });
})
