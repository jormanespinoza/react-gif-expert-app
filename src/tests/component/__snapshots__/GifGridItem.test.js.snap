// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGridItem should render <GifGridItem /> properly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="test"
    src="test_url"
  />
  <p>
    test
  </p>
</div>
`;
